### DISCUSSION TITLE
Stuck when editing past a .gitignore change

### DISCUSSION THREAD

**Discussion by evmar** [jj-vcs/jj#5194] [Q&A] [answered]
I have a repo where I added `node_modules` (full of big node goop) to `.gitignore`.

Suppose I `jj new zz` (where `zz` is my root commit).  `node_modules` is no longer ignored.

Now suppose I realize I made a mistake.  How can I recover?  Any `jj` command I run, including `jj undo` or `jj abandon` fails with

```
Failed to snapshot the working copy
The file [...node_modules goop] is too large to be snapshotted
```

It feels there should be some hint text offering to e.g. force whatever command you just attempted.

**Comment by scott2000** [comment, marked as answer]
`jj undo --config-toml 'snapshot.auto-track = "none()"'` might work? I think I did something similar to recover from a situation like this.

**Reply by evmar** [reply]
This worked!  But also I am not sure how I would ever have discovered it...

**Reply by jennings** [reply]
This question gets asked regularly, it would be wonderful if you (@evmar) wanted to [contribute it to the FAQ](https://github.com/jj-vcs/jj/blob/main/docs/FAQ.md).

**Reply by evmar** [reply]
There are some suggestions printed when this fails.  I wonder if this suggestion (or a FAQ link?) would be appropriate to print there too?

**Reply by evmar** [reply]
I'm gonna mark this as the answer even though I feel like there's a larger/better solution lurking...

**Comment by arxanas** [comment]
In principle, I believe you can add the ignore pattern again to `.git/info/exclude` to work around this, although I don't think it's a better solution than `--config-toml 'snapshot.auto-track = "none()"'` above.
